# Declare some variables as const to silence the compiler
#
# Copyright (C) 2006 - 2007 Nokia Corporation.
# This file is distributed under the terms of GNU LGPL license, either version 2
# of the License, or (at your option) any later version.
#
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml.c gtkhtml-3.24.4/gtkhtml/gtkhtml.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml.c gtkhtml-3.24.4/gtkhtml/gtkhtml.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml.c gtkhtml-3.24.4/gtkhtml/gtkhtml.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml.c gtkhtml-3.24.4/gtkhtml/gtkhtml.c
--- gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:29:01.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:29:01.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:29:01.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:29:01.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml.c	2007-02-16 16:44:08.000000000 +0200
@@ -92,7 +92,7 @@
 	DND_TARGET_TYPE_STRING,
 };
 
-static GtkTargetEntry dnd_link_sources [] = {
+static const GtkTargetEntry dnd_link_sources [] = {
 	{ "message/rfc822", 0, DND_TARGET_TYPE_MESSAGE_RFC822 },
 	{ "x-uid-list", 0, DND_TARGET_TYPE_X_UID_LIST },
 	{ "text/uri-list", 0, DND_TARGET_TYPE_TEXT_URI_LIST },
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c
diff -ru gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c
--- gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-12 08:57:38.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-12 08:57:38.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-12 08:57:38.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-12 08:57:38.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/gtkhtml-properties.c	2007-02-16 16:44:08.000000000 +0200
@@ -136,7 +136,7 @@
 
 /* enums */
 
-static GEnumValue _gtk_html_cursor_skip_values[] = {
+static const GEnumValue _gtk_html_cursor_skip_values[] = {
 	{ GTK_HTML_CURSOR_SKIP_ONE,  "GTK_HTML_CURSOR_SKIP_ONE",  "one" },
 	{ GTK_HTML_CURSOR_SKIP_WORD, "GTK_HTML_CURSOR_SKIP_WORD", "word" },
 	{ GTK_HTML_CURSOR_SKIP_PAGE, "GTK_HTML_CURSOR_SKIP_WORD", "page" },
@@ -155,7 +155,7 @@
 	return cursor_skip_type;
 }
 
-static GEnumValue _gtk_html_command_values[] = {
+static const GEnumValue _gtk_html_command_values[] = {
 	{ GTK_HTML_COMMAND_UNDO,  "GTK_HTML_COMMAND_UNDO",  "undo" },
 	{ GTK_HTML_COMMAND_REDO,  "GTK_HTML_COMMAND_REDO",  "redo" },
 	{ GTK_HTML_COMMAND_COPY,  "GTK_HTML_COMMAND_COPY",  "copy" },
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-09 21:54:58.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-09 21:54:58.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-09 21:54:58.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-09 21:54:58.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cursor.c	2007-02-16 16:44:08.000000000 +0200
@@ -35,18 +35,18 @@
 
 #define BLINK_TIMEOUT 500
 
-static GdkColor table_stipple_active_on      = { 0, 0,      0,      0xffff };
-static GdkColor table_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
-static GdkColor table_stipple_non_active_on  = { 0, 0xaaaa, 0xaaaa, 0xaaaa };
-static GdkColor table_stipple_non_active_off = { 0, 0xffff, 0xffff, 0xffff };
-
-static GdkColor cell_stipple_active_on      = { 0, 0x7fff, 0x7fff, 0 };
-static GdkColor cell_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
-static GdkColor cell_stipple_non_active_on  = { 0, 0x7aaa, 0x7aaa, 0x7aaa };
-static GdkColor cell_stipple_non_active_off = { 0, 0xffff, 0xffff, 0xffff };
+static const GdkColor table_stipple_active_on      = { 0, 0,      0,      0xffff };
+static const GdkColor table_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
+static const GdkColor table_stipple_non_active_on  = { 0, 0xaaaa, 0xaaaa, 0xaaaa };
+static const GdkColor table_stipple_non_active_off = { 0, 0xffff, 0xffff, 0xffff };
+
+static const GdkColor cell_stipple_active_on      = { 0, 0x7fff, 0x7fff, 0 };
+static const GdkColor cell_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
+static const GdkColor cell_stipple_non_active_on  = { 0, 0x7aaa, 0x7aaa, 0x7aaa };
+static const GdkColor cell_stipple_non_active_off = { 0, 0xffff, 0xffff, 0xffff };
 
-static GdkColor image_stipple_active_on      = { 0, 0xffff, 0,      0 };
-static GdkColor image_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
+static const GdkColor image_stipple_active_on      = { 0, 0xffff, 0,      0 };
+static const GdkColor image_stipple_active_off     = { 0, 0xffff, 0xffff, 0xffff };
 
 void
 html_engine_hide_cursor  (HTMLEngine *engine)
@@ -110,7 +110,7 @@
 
 static void
 draw_cursor_rectangle (HTMLEngine *e, gint x1, gint y1, gint x2, gint y2,
-		       GdkColor *on_color, GdkColor *off_color,
+		       const GdkColor *on_color, const GdkColor *off_color,
 		       gint offset)
 {
 	GdkGC *gc;
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-01-03 17:19:57.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlengine-edit-cut-and-paste.c	2007-02-16 16:44:08.000000000 +0200
@@ -1253,9 +1253,9 @@
 	html_engine_thaw (e);
 }
 
-static char *picto_chars = "DO)(|/PQ\0:-\0:\0:-\0:\0:;=-\0:;\0:-~\0:\0:\0:-\0:\0:-\0:\0:-\0:\0:-\0:\0";
-static gint picto_states [] = { 9, 14, 19, 27, 35, 40, 45, 50, 0, -1, 12, 0, -1, 0, -2, 17, 0, -2, 0, -3, -4, -5, 24, 0, -3, -4, 0, -6, 31, 33, 0, -6, 0, -11, 0, -8, 38, 0, -8, 0, -9, 43, 0, -9, 0, -10, 48, 0, -10, 0, -12, 53, 0, -12, 0};
-static gchar *picto_icon_names [] = {
+static const char *picto_chars = "DO)(|/PQ\0:-\0:\0:-\0:\0:;=-\0:;\0:-~\0:\0:\0:-\0:\0:-\0:\0:-\0:\0:-\0:\0";
+static const gint picto_states [] = { 9, 14, 19, 27, 35, 40, 45, 50, 0, -1, 12, 0, -1, 0, -2, 17, 0, -2, 0, -3, -4, -5, 24, 0, -3, -4, 0, -6, 31, 33, 0, -6, 0, -11, 0, -8, 38, 0, -8, 0, -9, 43, 0, -9, 0, -10, 48, 0, -10, 0, -12, 53, 0, -12, 0};
+static const gchar *picto_icon_names [] = {
	"stock_smiley-6",
	"stock_smiley-5",
	"stock_smiley-1",
diff -ru gtkhtml-3.24.4/gtkhtml/htmlentity.c gtkhtml-3.24.4/gtkhtml/htmlentity.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlentity.c gtkhtml-3.24.4/gtkhtml/htmlentity.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlentity.c gtkhtml-3.24.4/gtkhtml/htmlentity.c
diff -ru gtkhtml-3.24.4/gtkhtml/htmlentity.c gtkhtml-3.24.4/gtkhtml/htmlentity.c
--- gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-01-03 17:19:57.000000000 +0200
--- gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-01-03 17:19:57.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-02-16 16:44:08.000000000 +0200
+++ gtkhtml-3.24.4/gtkhtml/htmlentity.c	2007-02-16 16:44:08.000000000 +0200
@@ -37,7 +37,10 @@
 };
 typedef struct _EntityEntry EntityEntry;
 
-static EntityEntry entity_table[] = {
+/* FIXME: why not just sort these and use bsearch() instead of creating
+ * an additional glib hash table?
+ */
+static const EntityEntry entity_table[] = {
 
 	/* Latin1 */
 	{ 160,	"nbsp" },
